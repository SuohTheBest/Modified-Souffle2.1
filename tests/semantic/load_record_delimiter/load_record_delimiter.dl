// Souffle - A Datalog Compiler
// Copyright (c) 2020, The Souffle Developers. All rights reserved
// Licensed under the Universal Permissive License v 1.0 as shown at:
// - https://opensource.org/licenses/UPL
// - <souffle root>/licenses/SOUFFLE-UPL.txt

// Test reading where tuple delimiters coincide with record delimiter

.type pair = [f: number, t: symbol]


.decl A(x: pair, y: pair, z: pair)
.input A(delimiter=",")

.decl B(x: pair, y: pair, z: pair)
.input B(delimiter=", ")

.decl C(x: pair, y: pair, z: pair)
.input C(delimiter=" , ")

.decl O(x: pair, y: pair, z: pair)
O(x, y, z) :- A(x, y, z); B(x, y, z); C(x, y, z).

.output O(IO=stdout, delimiter=", ")
