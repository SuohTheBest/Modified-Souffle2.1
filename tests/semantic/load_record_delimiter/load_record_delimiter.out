---------------
O
===============
nil, nil, nil
nil, [2, Aapple], [2, Aapple]
nil, [2, Bapple], [2, Bapple]
nil, [2, Capple], [2, Capple]
[1, Ameow], [0, AmeowMeow], nil
[-3, Ablue], nil, [1, Ameow]
[1, Bmeow], [0, BmeowMeow], nil
[-3, Bblue], nil, [1, Bmeow]
[1, Cmeow], [0, CmeowMeow], nil
[-3, Cblue], nil, [1, Cmeow]
===============
